rfp-proposals/rfp-proposal-reputation-system.md

```diff
@@ -0,0 +1,326 @@
+        * % deviation of storage price from: 
+            * mean of all storage prices on the network. 
+            * median of all the storage prices of the network. 
+        * Reason: Some miners may be overpricing/underpricing their storage with respect to the market. 
+    * % of miner storage being used.  
+        * Formula = used storage/total storage. 
+        * Reason: Miners consistently hovering around > 80% may signal more trustworthiness. 
+    * Uptime:
+        * Reason: This is a more generic metric to track then failed deals. A miner may be up consistently but not getting any deals due to other reasons. 
+    * Miner Proactiveness: 
+        * Delta of time since Filecoin published an upgrade for node software and the miner updated itself. 
+        * Reason: 
+            * Smaller the delta, the more proactive the miner. 
+            * Miners who don't update are risky. 
+
+
+   
+### Specifics of web app: 
+
+
+#### Home page:
+Displaying  list of miners currently providing space for allocation with the following info:
+* Miner
+* Miner Score
+* Used Storage
+* Available storage
+* Storage Price
+* Valid till
+* Status
+
+ 
+![](https://i.imgur.com/OsVK6v2.png)
+
+^ Early Prototype. 
+
+<br><br>
+
+#### Miner Individual page:
+* Miner stats card containing information:
+    * Miner Address
+    * Filecoin version
+    * Collateral 
+    * Miner geographic location
+    * Storage Metrics: Used/Available/Total Storage. 
+
+* Miner Score Table: (10 days average):
+    * [Final/Average Score](#How-will-scores-be-calculated?)
+    * Individual characteristics score:
+        * Latency
+        * % of successful deals
+        *  %  of slashed and dropped contracts
+        * etc
+
+* Comparison with storage miners with the same score:
+    * Storage price
+    * Collateral
+    * Est. uptime
+    * Stored data
+
+* Substantiated feedback:
+    * identifying the user:
+        * a message-signing-based authentication mechanism with a user's public address as their identifier.
+    * identifying the event:
+        * enter dealid/(id associated with the event): 
+        * Choose check event:
+        * Comment/feedback
+        * Submit
+        * Post if the event is verified.
+
+
+![](https://i.imgur.com/zDz75Bd.png)
+
+^ Early Prototype. 
+
+
+
+## Development Roadmap
+
+Tentative Start: Feb 1, 2019
+### Milestone 1: Get vantage on: 
+
+1. Scope of the reputation system service: 
+    * Collaborate with FileCoin DevGrants team to: 
+        * Gain feedback on the metrics we have thought of. How relevant would they be for users now?  
+        * Gain awareness on the kinds of objective feedback users will give that can be verifiable from the service.
+    * Iron out reputation score logic that will be the single metric that all miners will be ranked upon. 
+    * Finalize the reputation system service requirements spec sheet.
+     
+2. and how the tool would be built: 
+    * Syncup process from Filecoin Network to DB. 
```

Comment:
Hi @sahilnanda1995,

Question: What database do you intend to use?

Reply:
Hi @eshon, 

At the moment, we intend to go forward with postgres. In my research, I came across the  ChainWatch tool which now seems to be configured with postgres as well.  Proposed path is thinking how much overlap exists so we can leverage existing work, and how much extra features we would need to add for the service to be complete in its feature set.